content/sensu-go/6.8/catalog/build-private-catalog.md

```diff
@@ -0,0 +1,191 @@
+  - filters: ...
+{{< /code >}}
+
+If needed, replace `assets.bonsai.sensu.io` with your preferred URL in asset `builds.url` values in all `sensu-resources.yaml` files before you continue.
+
+## Install the catalog-api command line tool
+
+The Sensu Catalog API includes the catalog-api command line tool.
+The catalog-api tool is an open-source static API generator: it renders static HTTP API content that the Sensu web UI can consume.
+
+To install the catalog-api tool:
+
+1. Clone the Sensu Catalog API repository:
+{{< code shell >}}
+git clone https://github.com/sensu/catalog-api
+{{< /code >}}
+
+2. Navigate to the local catalog-api repository:
+{{< code shell >}}
+cd catalog-api
+{{< /code >}}
+
+3. Build the `catalog-api` tool:
+{{< code shell >}}
+go build
+{{< /code >}}
+
+## Clone and validate the integration repository
+
+The catalog-api tool consumes content from a repository that includes all the files required to build a catalog of integrations.
+Follow these steps to clone your repository and validate that all files are organized properly:
+
+1. Clone the repository that stores your Sensu integrations.
+This example uses Sensu's public integration repository:
+{{< code shell >}}
+git clone https://github.com/sensu/catalog
+{{< /code >}}
+
+2. Navigate to your local copy of the repository that stores the Sensu integrations.
+This example uses https://github.com/sensu/catalog, so the repository is `catalog`: 
+{{< code shell >}}
+cd ../catalog
+{{< /code >}}
+
+3. Validate the integration repository contents:
+{{< code shell >}}
+../catalog-api/catalog-api catalog validate --repo-dir .
+{{< /code >}}
+
+## Generate the private catalog
+
+With a validated repository, you can generate your private catalog locally:
+
+{{< code shell >}}
+../catalog-api/catalog-api catalog generate
+{{< /code >}}
+
+The `generate` subcommand generates the static API in a temporary directory, `/tmp/generated-api/`.
+
+These catalog builds are versioned so that every previous iteration is available.
+
+The catalog-api tool generates builds into a checksum-based output directory structure.
+The path to the latest or production Catalog API content is managed by a `version.json` file (in this example, the file would be https://catalog.sensu.io/version.json) that instructs the Sensu web UI to load API contents from the specified checksum subdirectory.
+For example:
+
+{{< code text >}}
+{
+  "release_sha256": "d0e2ba810c3d546c82121406fb5f214e66aeb7fe9026706f9a7391463cf4da19",
+  "last_updated": 1657574563
+}
+{{< /code >}}
+
+## Publish the static API to an endpoint
+
+Once you have generated the static API, you can serve the output on any HTTP service and publish it to any endpoint.
+
+The only requirement is that the endpoint URL must be fetchable for your web UI users.
+The web UI fetches catalog content from your endpoint; the Sensu backend does not serve any of the catalog content.
```

Comment:
Is there a catalog-api command for this? it's not clear to me how a user goes from generating the catalog with catalog-api to having the catalog content published at an endpoint.

Reply:
Maybe instead of an explanation we should use Sensu's process as an example. In a demo video, Caleb describes it as something like generating the output, copying the output to an S3 bucket, and using Cloudfront CDN to serve the content to https://catalog.sensu.io.